pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -305,37 +305,68 @@ type VitessShardStatus struct {
 	// Cells is a list of cells in which any tablets for this shard are deployed.
 	Cells []string `json:"cells,omitempty"`
 
-	// HasMaster is a condition indicating whether the Vitess topology
-	// reflects a master for this shard.
-	HasMaster corev1.ConditionStatus `json:"hasMaster,omitempty"`
-	// HasInitialBackup is a condition indicating whether the initial backup
-	// has been seeded for the shard.
-	HasInitialBackup corev1.ConditionStatus `json:"hasInitialBackup,omitempty"`
+	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
+	Conditions map[VitessShardConditionType]*VitessShardCondition `json:"conditions,omitempty"`
+
 	// MasterAlias is the tablet alias of the master according to the global
 	// shard record. This could be empty either because there is no master,
 	// or because the shard record could not be read. Check the HasMaster
 	// condition whenever the distinction is important.
 	MasterAlias string `json:"masterAlias,omitempty"`
 
-	// Idle is a condition indicating whether the shard can be turned down.
+	// BackupLocations reports information about the backups for this shard in
+	// each backup location.
+	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+}
+
+// VitessShardConditionType is a valid value for VitessShardCondition.Type
+type VitessShardConditionType string
+
+// These are valid conditions of VitessShard.
+const (
+	// HasMaster indicates whether the Vitess topology reflects a master for this shard.
+	HasMaster VitessShardConditionType = "HasMaster"
+	// HasInitialBackup indicates whether the initial backup has been seeded for the shard.
+	HasInitialBackup VitessShardConditionType = "HasInitialBackup"
+	// Idle indicates whether the shard can be turned down.
 	// If Idle is True, the shard is not part of the active shard set
 	// (partitioning) for any tablet type in any cell, so it should be safe
 	// to turn down the shard.
-	Idle corev1.ConditionStatus `json:"idle,omitempty"`
+	Idle VitessShardConditionType = "Idle"
+)
 
-	// BackupLocations reports information about the backups for this shard in
-	// each backup location.
-	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+// VitessShardCondition contains details for the current condition of this VitessShard.
+type VitessShardCondition struct {
+	// Type is the type of the condition.
+	Type VitessShardConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=VitessShardConditionType"`
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=ConditionStatus"`
+	// Last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty" protobuf:"bytes,3,opt,name=lastProbeTime"`
```

Comment:
Omit this field. It is deprecated and not recommended for new objects because it's too noisy.

Reply:
Will do.

Reply:
Removed https://github.com/planetscale/vitess-operator/pull/31/commits/4d12ec3679395986ef1d9ffbec7fd3feee478a36.